File: II semestr/HomeWork2/Zadacha2 List/Zadacha2 List/List.cs
namespace ListNamespace
{
    using System;

    public class List
    {
        private class ListElement
        {
            private int eValue;
            public ListElement Next { get; set; }

            /// <summary>
            /// Getter and setter for value.
            /// </summary>
            public int Value { get; set; }
        }

        private ListElement head;

        /// <summary>
        /// Insert value in the head of the list.
        /// </summary>
        /// <param name="value"></param>
        public void InsertToHead(int value)
        {
            var newElement = new ListElement()
            {
                Value = value,
                Next = head
            };
            head = newElement;
        }

        /// <summary>
        /// Return value of the position.
        /// </summary>
        /// <param name="position"></param>
        /// <returns></returns>
        public int ValueOfPosition(int position)
        {
            var element = head;
            if (element == null)
            {
                return -1;
            }
            for (int i = 0; i < position; i++)
            {
                if (element == null)
                {
                    return -1;
                }
                element = element.Next;
            }
            return element.Value;
        }

        /// <summary>
        /// Print all list on console.
        /// </summary>
        public void Print()
        {
            var element = head;
            while (element != null)
            {
                Console.WriteLine("Element = {0}", element.Value);
                element = element.Next;
            }
        }

        /// <summary>
        /// Clean all list.
        /// </summary>
        public void RemoveList()
        {
            head = null;
        }

        /// <summary>
        /// Return 'true' if list is empty and 'false' if it is not empty.
        /// </summary>
        /// <returns></returns>
        public bool IsEmpty()
        {
            return (head == null);
        }

        /// <summary>
        /// Is value in the list?
        /// </summary>
        /// <param name="value"></param>
        /// <returns></returns>
        public bool Exist(int value)
        {
            var element = head;
            while (element != null)
            {
                if (element.Value == value)
                {
                    return true;
                }
                element = element.Next;
            }
            return false;
        }
    }
}

File: I semestr/KR2/Zadacha 1/KR2Zadacha1/list.cpp
#include <iostream>

#include "list.h"

; using namespace std;

struct ListElement
{
	ElementType value;
	ListElement *next;
};

struct List
{
	ListElement *head;
};

List *create()
{
	List *result = new List;
	result->head = NULL;
	return result;
}

void deleteList(List *list)
{
	Position position = list->head;
	while (position != nullptr)
	{
		Position temp = position;
		position = position->next;
		delete temp;
	}
	delete position;
	delete list;
}

void insert(List *list, ElementType value)
{
	Position position = list->head;
	if (position == nullptr)
	{
		insertToHead(list, value);
		return;
	}
	while (position->next != nullptr)
	{
		position = position->next;
	}
	ListElement *newElement = new ListElement;
	newElement->value = value;
	newElement->next = position->next;
	position->next = newElement;
}

void insertToHead(List *list, ElementType value)
{
	ListElement *newElement = new ListElement;
	newElement->value = value;
	newElement->next = list->head;
	list->head = newElement;
}

Position first(List *list)
{
	return list->head;
}

Position end(List *list)
{
	return nullptr;
}

Position next(List *list, Position position)
{
	return position->next;
}

ElementType valueOnPosition(List *list, Position position)
{
	return position->value;
}